File: 8kyu/get-planet-name-by-id/get-planet-name-by-id.py
# def get_planet_name(id):
#     name=""
#     match id:
#         case 1: name = "Mercury"
#         case 2: name = "Venus"
#         case 3: name = "Earth"
#         case 4: name = "Mars"
#         case 5: name = "Jupiter"
#         case 6: name = "Saturn"
#         case 7: name = "Uranus"  
#         case 8: name = "Neptune"
#     return name

def get_planet_name(id):
  planets = {
    1: "Mercury",
    2: "Venus",
    3: "Earth",
    4: "Mars",
    5: "Jupiter",
    6: "Saturn",
    7: "Uranus", 
    8: "Neptune",
  }
  return planets[id]


# write unit test to test get_planet_name function
def test_get_planet_name():
    assert get_planet_name(1) == "Mercury"
    assert get_planet_name(2) == "Venus"
    assert get_planet_name(3) == "Earth"
    assert get_planet_name(4) == "Mars"
    assert get_planet_name(5) == "Jupiter"
    assert get_planet_name(6) == "Saturn"
    assert get_planet_name(7) == "Uranus"  
    assert get_planet_name(8) == "Neptune"
    

File: 8kyu/get-planet-name-by-id/get-planet-name-by-id.go
package main

import "fmt"

// func GetPlanetName(ID int) string {
// 	switch ID {
// 		case 1:
// 			return "Mercury"
// 		case 2:
// 			return "Venus"
// 		case 3:
// 			return "Earth"
// 		case 4:
// 			return "Mars"
// 		case 5:
// 			return "Jupiter"
// 		case 6:
// 			return "Saturn"
// 		case 7:
// 			return "Uranus"
// 		case 8:
// 			return "Neptune"
// 		default:
// 			return "Pluto" // ;-)
// 	}
// }

func GetPlanetName(ID int) string {
	result := []string{"Mercury","Venus","Earth","Mars","Jupiter","Saturn","Uranus","Neptune","Pluto"}
	return result[ID-1]
}

func main() {
	fmt.Println(GetPlanetName(1))
	fmt.Println(GetPlanetName(3))
}